hack/download-restic.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# Copyright 2020 the Velero contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [[ -z "${BIN}" ]]; then
+    echo "BIN must be set"
+    exit 1
+fi
+
+if [[ "${BIN}" != "velero" ]]; then
```

Comment:
In the old setup, the `restic` binary was only downloaded into the `Docker-velero[-$ARCH]` images. This is why I am checking if the current binary being built is velero, if not, it will not download restic. I don't mind changing stuff about this, as I was just mimicking what was included before to ensure 100% compatibility.

Reply:
Example of before: https://github.com/vmware-tanzu/velero/blob/main/Dockerfile-velero#L21

The reason I am doing it like this now is so we can use a single `Dockerfile` for all the architectures and binaries. Without this check, the restic binary would also end up in the `velero-restic-restore-helper` image, which it wasn't in before. It probably would not break anything, but this was just me making sure the end-result is still 100% the same :)